File: mylibrary/src/main/java/com/giosis/library/message/MessageListAdapter.java
package com.giosis.library.message;

import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.os.Build;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.giosis.library.R;

import java.util.ArrayList;

public class MessageListAdapter extends BaseAdapter {

    private final Context context;
    String called_fragment;

    RelativeLayout layout_message_list_item;
    TextView text_message_list_item_title;
    TextView text_message_list_item_message;
    TextView text_message_list_item_date;


    private final ArrayList<MessageListResult.MessageList> mItems;

    public MessageListAdapter(Context context, String called, ArrayList<MessageListResult.MessageList> messageItemArrayList) {

        this.context = context;           // getActivity()
        this.called_fragment = called;
        this.mItems = messageItemArrayList;
    }


    @Override
    public int getCount() {
        if (mItems != null && mItems.size() > 0) {
            return mItems.size();
        }

        return 0;
    }


    @Override
    public Object getItem(int position) {
        return mItems.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(final int position, View convertView, ViewGroup parent) {

        View view;

        if (convertView == null) {

            LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            view = inflater.inflate(R.layout.item_message_list, null);
        } else {

            view = convertView;
        }

        layout_message_list_item = view.findViewById(R.id.layout_message_list_item);
        text_message_list_item_title = view.findViewById(R.id.text_message_list_item_title);
        text_message_list_item_message = view.findViewById(R.id.text_message_list_item_message);
        text_message_list_item_date = view.findViewById(R.id.text_message_list_item_date);


        if (mItems.get(position).getRead_yn().equals("Y")) {

            text_message_list_item_message.setTextColor(Color.parseColor("#444444"));
        } else {

            text_message_list_item_message.setTextColor(Color.parseColor("#f22020"));
        }

        if (called_fragment.equalsIgnoreCase("C")) {     // Customer

            text_message_list_item_title.setText(mItems.get(position).getTracking_no());
        } else if (called_fragment.equalsIgnoreCase("A")) {  // Admin

            text_message_list_item_title.setText(mItems.get(position).getSender_id());
        }


        if (Build.VERSION.SDK_INT >= 24) {

            text_message_list_item_message.setText(Html.fromHtml(mItems.get(position).getMessage(), Html.FROM_HTML_MODE_LEGACY));
        } else {

            text_message_list_item_message.setText(Html.fromHtml(mItems.get(position).getMessage()));
        }

        text_message_list_item_date.setText(mItems.get(position).getTime(called_fragment));


        layout_message_list_item.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (called_fragment.equalsIgnoreCase("C")) {     // Customer

                    Intent intent = new Intent(context, CustomerMessageListDetailActivity.class);
                    intent.putExtra("question_no", mItems.get(position).getQuestion_seq_no());
                    intent.putExtra("tracking_no", mItems.get(position).getTracking_no());
                    context.startActivity(intent);
                } else if (called_fragment.equalsIgnoreCase("A")) {  // Admin

                    Intent intent = new Intent(context, AdminMessageListDetailActivity.class);
                    intent.putExtra("sender_id", mItems.get(position).getSender_id());
                    context.startActivity(intent);
                }
            }
        });

        return view;
    }
}

File: mylibrary/src/main/java/com/giosis/library/message/MessageListAdapter.kt
package com.giosis.library.message

import android.content.Intent
import android.graphics.Color
import android.os.Build
import android.text.Html
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.giosis.library.databinding.ItemMessageListBinding
import java.util.*

class MessageListAdapter(private var called: String, private val mItems: ArrayList<MessageListResult>?) : RecyclerView.Adapter<MessageListAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {

        val binding = ItemMessageListBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    class ViewHolder(val binding: ItemMessageListBinding) : RecyclerView.ViewHolder(binding.root)

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        if (mItems!![position].read_yn == "Y") {
            holder.binding.textMessage.setTextColor(Color.parseColor("#444444"))
        } else {
            holder.binding.textMessage.setTextColor(Color.parseColor("#f22020"))
        }

        if (called == "C") {     // Customer
            holder.binding.textTitle.text = mItems[position].tracking_no
        } else if (called == "A") {  // Admin
            holder.binding.textTitle.text = mItems[position].sender_id
        }


        if (Build.VERSION.SDK_INT >= 24) {

            holder.binding.textMessage.text = Html.fromHtml(mItems[position].message, Html.FROM_HTML_MODE_LEGACY)
        } else {
            holder.binding.textMessage.text = Html.fromHtml(mItems[position].message)
        }

        holder.binding.textDate.text = mItems[position].getTime(called)


        holder.binding.layoutMessageItem.setOnClickListener {
            if (called == "C") {     // Customer

                val intent = Intent(it.context, CustomerMessageListDetailActivity::class.java)
                intent.putExtra("question_no", mItems[position].question_seq_no)
                intent.putExtra("tracking_no", mItems[position].tracking_no)
                it.context.startActivity(intent)
            } else if (called == "A") {  // Admin

                val intent = Intent(it.context, AdminMessageListDetailActivity::class.java)
                intent.putExtra("sender_id", mItems[position].sender_id)
                it.context.startActivity(intent)
            }
        }
    }

    override fun getItemCount(): Int {
        return if (mItems != null && mItems.size > 0) {
            mItems.size
        } else 0
    }
}